segmentation.py

```diff
@@ -0,0 +1,575 @@
+        else:
+            msg = "The input 'minprob' must be a float greater than zero."
+            raise Exception(msg)
+
+        # Allocate variables for tracking segments.
+        self.__hypotheses = list()
+        self.__index = list()
+        self.__probabilities = list()
+
+        self.__featfun = featfun if callable(featfun) else lambda x: x
+        self.__ratefun = ratefun if callable(ratefun) else lambda x: ratefun
+
+    def __initialise_algorithm(self, m, n):
+        """Initialise the Bcdm algorithm."""
+
+        # Ensure input dimensions are consistent.
+        if self.__m is None:
+            self.__m = m
+        elif self.__m != m:
+            msg = 'Expected %i dimensions in the predictor variable.' % m
+            raise Exception(msg)
+
+        # Ensure output dimensions are consistent.
+        if self.__n is None:
+            self.__n = n
+        elif self.__n != n:
+            msg = 'Expected %i dimensions in the response variable.' % n
+            raise Exception(msg)
+
+        # Set uninformative prior for the location parameter.
+        if self.__mu is None:
+            self.__mu = np.zeros([m, n])
+
+        # Set uninformative prior for the scale parameter.
+        if self.__omega is None:
+            self.__omega = np.eye(m)
+
+        # Set uninformative prior for the dispersion/noise parameter.
+        if self.__sigma is None:
+            self.__sigma = np.eye(n)
+
+        # Set uninformative prior for the shape parameter.
+        if self.__eta is None:
+            self.__eta = n
+
+        stat = MatrixVariateNormalInvGamma(self.__mu,
+                                           self.__omega,
+                                           self.__sigma,
+                                           self.__eta)
+
+        # Create the initial hypothesis, which states that the first segment is
+        # about to begin.
+        self.__update_count = 0
+        self.__hypotheses = [{'index': 0,
+                              'count': 0,
+                              'log_probability': 0.0,
+                              'distribution': stat,
+                              'log_constant': stat.logconst()}]
+
+    def __accum(self, x, y):
+        return max(x, y) + np.log1p(np.exp(-abs(x - y)))
+
+    def block_update(self, X, Y):
+        """Update model with multiple observations.
+
+        This method is a convenience method for updating the model when many
+        new input-output data are available. This method simply iterates over
+        the inputs calling :py:meth:`.update`.
+
+        Args:
+            X (numpy.array): Observed (K x M) input data (predictor variable).
+            Y (numpy.array): Observed (K x N) output data (response variable).
+
+        """
+
+        # Iterate through the rows of the input-output data updating the
+        # hypotheses in the model.
+        for i in range(X.shape[0]):
+            self.update(X[i, :], Y[i, :])
```

Comment:
Ahh... yes. Those are two very different things. Correct, I don't believe it crashed but the results were indeed meaningless.  This is why I wrote `block_update()` originally. It was just to stop users from having to write an external loop. Loops are fairly compact in python so removing the method is not a big inconvenience to users (it could be added as an option to `update()`?). Though we should make a note in the documentation that the resolution of the segmentation is linked to how many data points are inserted during calls to `update()`.